src/app/route/ProductPage/ProductPage.component.js

```diff
@@ -172,24 +172,36 @@ class ProductPage extends Component {
 
     /**
      * Update query params without adding to history, set configurableVariantIndex
-     * @param {Number} variant
+     * @param {string} key
+     * @param {number|string} value
      */
-    updateUrl(variant) {
-        const { configurableVariantIndex } = this.state;
+    updateUrl(key, value) {
+        const { product: { variants, configurable_options } } = this.props;
+        const { configurableVariantIndex, parameters: oldParameters } = this.state;
 
-        if (configurableVariantIndex !== variant) updateQueryParamWithoutHistory('variant', variant);
+        const parameters = {
+            ...oldParameters,
+            [key]: `${value}`
+        };
 
-        return this.setState({ configurableVariantIndex: variant });
+        this.setState({ parameters });
+        updateQueryParamWithoutHistory(key, value);
+
+        const newIndex = getVariantIndex(variants, parameters);
+        if (Object.keys(parameters).length === Object.keys(configurable_options).length
```

Comment:
Why is this check needed here? Could not we check for `configurableVariantIndex !== newIndex` only?

Reply:
Without `(Object.keys(parameters).length === Object.keys(configurable_options).length` check, AddToCart button becomes enabled when there is only one parameter selected and user switches it to another parameter of the same type (e.g. no color & size 34 -> size 36)